public/app/core/components/NavBar/Next/NavBarItem.tsx

```diff
@@ -113,6 +113,7 @@ const NavBarItem = ({
                     url={item.url}
                     onClick={item.onClick}
                     styleOverrides={cx(styles.primaryText, { [styles.header]: isSection })}
+                    showIconInNavbar={item.showIconInNavbar}
```

Comment:
i wonder if we can just use this to conditionally pass the icon on line 110 instead? something like:
```typescript
icon={icon.showIconInNavbar && !isSection ? (item.icon as IconName) : undefined}`
```

then `NavBarMenuItem` doesn't need a new prop at all.

Reply:
🤔 it might work, but tbh that line looks harder to read 🙈

Reply:
That's a good point ☺️